### PR TITLE
Octokit 3.0.0

### PR DESCRIPTION
**Pull request by alexottoboni**
Upgrades Octokit from version 1.22 to version 3.0.0 and fixes dependency issues
with faraday and faraday middleware.

### CODE REVIEW

File: git-scripts.gemspec [hunk 1]
```diff
@@ -20,8 +20,8 @@ Gem::Specification.new do |s|
    s.add_dependency 'json', '~> 1.8.0'
    s.add_dependency 'multi_json', '= 1.8.0'
    s.add_dependency 'highline', '= 1.6.19'
-   s.add_dependency 'faraday', '= 0.8.8'
-   s.add_dependency 'faraday_middleware', '= 0.9.0'
+   s.add_dependency 'faraday'
+   s.add_dependency 'faraday_middleware'
```
Comment (marczych) [inline git-scripts.gemspec]: Do we need to specify faraday at all now that octokit properly pulls in the correct version?

File: git-scripts.gemspec [hunk 2]
```diff
@@ -20,8 +20,8 @@ Gem::Specification.new do |s|
    s.add_dependency 'json', '~> 1.8.0'
```
Comment (marczych) [inline git-scripts.gemspec]: The comment directly above this isn't accurate anymore.

File: git-scripts.gemspec (hunk 1 above)
Comment (alexottoboni) [inline git-scripts.gemspec]: It looks like if we don't specify faraday or faraday_middleware it just installs faraday without faraday_middleware.

Comment (danielbeardsley) [inline git-scripts.gemspec]: I highly doubt that octokit v3 doesn't specify its dependencies correctly. I bet it'll work if you drop this and only specify the dependencies *we* need.

**Comment by danielbeardsley** [comment]
Far fewer changes than I realized!

File: git-scripts.gemspec (hunk 1 above)
Comment (alexottoboni) [inline git-scripts.gemspec]: It looks like Octokit 3 doesn't use faraday_middleware.

File: git-scripts.gemspec (hunk 2 above)
Comment (alexottoboni) [inline git-scripts.gemspec]: I removed it.

**Comment by marczych** [comment]
Wow, good for GitHub for being so backwards compatible across 2 major versions!

CR :ocean: 

This has the potential to break any users with plugins so we should bump the version accordingly.

**Comment by marczych** [comment]
CR :octocat:

**Comment by danielbeardsley** [comment]
CR :soon:

**Pull request merged**